datafusion/physical-expr/src/expressions/is_null.rs

```diff
@@ -100,6 +110,49 @@ impl PhysicalExpr for IsNullExpr {
     }
 }
 
+pub(crate) fn union_is_null(union_array: &UnionArray) -> Result<BooleanArray> {
+    if let Some(offsets) = union_array.offsets() {
+        dense_union_is_null(union_array, offsets)
+    } else {
+        sparce_union_is_null(union_array)
+    }
+}
+
+fn dense_union_is_null(
+    union_array: &UnionArray,
+    offsets: &ScalarBuffer<i32>,
+) -> Result<BooleanArray> {
+    let child_arrays = (0..union_array.type_names().len())
+        .map(|type_id| {
+            compute::is_null(&union_array.child(type_id as i8)).map_err(Into::into)
+        })
+        .collect::<Result<Vec<BooleanArray>>>()?;
+
+    let buffer: BooleanBuffer = offsets
+        .iter()
+        .zip(union_array.type_ids())
+        .map(|(offset, type_id)| child_arrays[*type_id as usize].value(*offset as usize))
+        .collect();
+
+    Ok(BooleanArray::new(buffer, None))
+}
+
+fn sparce_union_is_null(union_array: &UnionArray) -> Result<BooleanArray> {
```

Comment:
same.

Reply:
Nit: It may be a British vs American thing but I think typically an `s` is used: 

```suggestion
fn sparse_union_is_null(union_array: &UnionArray) -> Result<BooleanArray> {
```

This is also consistent with the arrow spec and library https://docs.rs/arrow/latest/arrow/array/struct.UnionArray.html

Reply:
done.